### PR TITLE
[Feature/PUR-173-pdpbreadtrack]: PDP breadcrumb track

### PR DESCRIPTION
**Pull request by kevin-ho87**
**Ticket**
https://aesoponline.atlassian.net/browse/PUR-173

**Description**
- Pass callback function to PDP header when a link in breadcrumb is clicked
- Data props in callback = breadcrumb clicked, selected product variant
- In Web-UI, this data will be used to submit datalayer push

### CODE REVIEW

**Review by MkMan — APPROVED** [review]

File: src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 1]
```diff
@@ -24,6 +25,7 @@ const Breadcrumbs: FC<BreadcrumbsProps> = ({
             <Hyperlink
               className={styles.link}
               id={item.id}
+              onClick={() => onHyperlinkClick(item)}
```
Comment (beau-charman) [inline src/components/Breadcrumbs/Breadcrumbs.tsx:28]: Can you bump this one out of the render for me? thank you!

Like:

```
const handleOnHyperlinkClick() => onHyperlinkClick(item);
...
onClick={handleOnHyperlinkClick}
```

File: src/components/ProductDetailHeader/ProductDetailHeader.js [hunk 2]
```diff
@@ -24,6 +35,9 @@ const ProductDetailHeader = ({ breadcrumbs, className, copy, theme }) => {
             <Breadcrumbs
               className={styles.breadcrumbs}
               items={breadcrumbs.items}
+              onHyperlinkClick={item =>
+                onBreadcrumbClick(item, selectedVariant)
```
Comment (beau-charman) [inline src/components/ProductDetailHeader/ProductDetailHeader.js:39]: Same here for the anonymous function in the render

File: src/components/ProductDetailHeader/ProductDetailHeader.js [hunk 3]
```diff
@@ -43,6 +57,9 @@ const ProductDetailHeader = ({ breadcrumbs, className, copy, theme }) => {
...
```
Comment (beau-charman) [inline src/components/ProductDetailHeader/ProductDetailHeader.js:61]: And here, thank you! (Same here for the anonymous function in the render)

**Review by beau-charman — CHANGES_REQUESTED** [review]
Just had a few items around inline anonymous function in the render returns

File: src/components/Breadcrumbs/Breadcrumbs.tsx (hunk 1 above)
Comment (beau-charman) [inline src/components/Breadcrumbs/Breadcrumbs.tsx:28]: For example, abstract out the `map`'s UI logic. 

```
const ListItem = ({ item }) => {
  const handleOnHyperlinkClick = () => onHyperlinkClick(item);
    
  return (
    <li className={styles.item} >
      <Hyperlink
        className={styles.link}
        id={item.id}
        onClick={handleOnHyperlinkClick}
        theme={theme}
        title={item.title}
        url={item.url}
      >
        {item.label}
      </Hyperlink>
    </li>
  )
}

return (
  <nav className={classSet} data-testid="data-testid-Breadcrumbs">
    <ul className={styles.list}>
      {items.map(item => (
        <ListItem item={item} key={item.id} />
      ))}
    </ul>
  </nav>
);
```

Comment (kevin-ho87) [reply src/components/Breadcrumbs/Breadcrumbs.tsx]: updated

File: src/components/ProductDetailHeader/ProductDetailHeader.js (hunk 2 above)
Comment (kevin-ho87) [reply src/components/ProductDetailHeader/ProductDetailHeader.js]: updated

File: src/components/Breadcrumbs/Breadcrumbs.tsx (hunk 1 above)
Comment (kevin-ho87) [reply src/components/Breadcrumbs/Breadcrumbs.tsx]: updated

**Review by beau-charman — APPROVED** [review]

**Comment by zack76** [comment]
Does it go to develop? @kevin-ho87

**Pull request merged**